### PR TITLE
Add documentation to note that some diagnostic plugins must be added after `DiagnosticsPlugin`

### PR DESCRIPTION
**Pull request by opstic**
# Objective

- Some diagnostic plugins do not currently have docs to explain that it must be added after `DiagnosticsPlugin`

## Solution

- Add documentation for it

### CODE REVIEW

File: crates/bevy_diagnostic/src/frame_time_diagnostics_plugin.rs [hunk 1]
```diff
@@ -4,7 +4,8 @@ use bevy_core::FrameCount;
 use bevy_ecs::prelude::*;
 use bevy_time::Time;
 
-/// Adds "frame time" diagnostic to an App, specifically "frame time", "fps" and "frame count"
+/// Adds "frame time" diagnostic to an [`App`], specifically "frame time", "fps" and "frame count"
+/// Must be added after [`DiagnosticsPlugin`]
```
Comment (alice-i-cecile) [inline crates/bevy_diagnostic/src/frame_time_diagnostics_plugin.rs:8]:
```suggestion
///
/// # Warning
///
/// Must be added after [`DiagnosticsPlugin`] or a panic will occur due to missing resources.
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Formatting / style nit + more clarity on why the panic occurs, then LGTM.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by mockersf** [comment]
If we add this doc for `FrameTimeDiagnosticsPlugin`, it should be added to all the other diagnostics plugins:
- `AssetCountDiagnosticsPlugin`
- `EntityCountDiagnosticsPlugin`
- `SystemInformationDiagnosticsPlugin`

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
see https://github.com/bevyengine/bevy/pull/8783#issuecomment-1585842232

**Review by alice-i-cecile — APPROVED** [review]

**Comment by opstic** [comment]
Closing in favor of #8819

**Pull request closed without merging**